x/feeburner/module.go

```diff
@@ -0,0 +1,197 @@
+
+// Deprecated: use RegisterServices
+func (am AppModule) LegacyQuerierHandler(_ *codec.LegacyAmino) sdk.Querier {
+	return nil
+}
+
+// RegisterServices registers a gRPC query service to respond to the module-specific gRPC queries
+func (am AppModule) RegisterServices(cfg module.Configurator) {
+	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.keeper))
+	types.RegisterQueryServer(cfg.QueryServer(), am.keeper)
+}
+
+// RegisterInvariants registers the invariants of the module. If an invariant deviates from its predicted value, the InvariantRegistry triggers appropriate logic (most often the chain will be halted)
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+// InitGenesis performs the module's genesis initialization. It returns no validator updates.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, gs json.RawMessage) []abci.ValidatorUpdate {
+	var genState types.GenesisState
+	// Initialize global index to index in genesis state
+	cdc.MustUnmarshalJSON(gs, &genState)
+
+	InitGenesis(ctx, am.keeper, genState)
+
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the module's exported genesis state as raw JSON bytes.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	genState := ExportGenesis(ctx, am.keeper)
+	return cdc.MustMarshalJSON(genState)
+}
+
+// ConsensusVersion is a sequence number for state-breaking change of the module. It should be incremented on each consensus-breaking change introduced by the module. To avoid wrong/empty versions, the initial version should be set to 1
+func (AppModule) ConsensusVersion() uint64 { return 1 }
+
+// BeginBlock contains the logic that is automatically triggered at the beginning of each block
+func (am AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
+
+// EndBlock contains the logic that is automatically triggered at the end of each block
+func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
+	moduleAddr := am.accountKeeper.GetModuleAddress(consumertypes.ConsumerRedistributeName)
+	if moduleAddr == nil {
+		// TODO: maybe panic or print warning?
+		return []abci.ValidatorUpdate{}
+	}
+
+	params := am.keeper.GetParams(ctx)
+
+	balance := am.bankKeeper.GetBalance(ctx, moduleAddr, params.NeutronDenom)
+	if !balance.IsZero() {
+		am.keeper.RecordBurnedFees(ctx, balance)
+		err := am.bankKeeper.BurnCoins(ctx, consumertypes.ConsumerRedistributeName, sdk.Coins{balance})
+		if err != nil {
+			// TODO: maybe panic or print warning?
+			return []abci.ValidatorUpdate{}
+		}
+	}
+
+	balance = am.bankKeeper.GetBalance(ctx, moduleAddr, params.GaiaDenom)
+	if !balance.IsZero() {
+		err := am.bankKeeper.SendCoins(
+			ctx,
+			moduleAddr,
+			sdk.MustAccAddressFromBech32(params.TreasuryAddress),
+			sdk.Coins{balance},
+		)
+		if err != nil {
+			// TODO: maybe panic or print warning?
```

Comment:
i think that we should panic here. @albertandrejev @pr0n00gler  wdyt?

Reply:
it will not break our tokenomics if it fails. Right now we do not have any scenario what to do with non-NTRN denoms, don't we? So I think we can print warning here.

Reply:
it will not break our tokenomics if it fails. Right now we do not have any scenario what to do with non-NTRN denoms, don't we? So I think we can print warning here.

Reply:
okay, let it be so